### PR TITLE
Adding support of raw format only as input

### PR DESCRIPTION
**Pull request by umishko**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by agraf** [comment]
Nice! Taking an efivarfs directory and turning it into binary UEFI variables is a great use case for this tool. Beware that this method only works for runtime variables - you lose boot time variable data because they are invisible after ExitBootServices. FWIW you also lose Secure Boot protection data, because you can not read the protections using GetVariable().

A few comments:

1. Please use a different name. "raw" is very ambiguous. How about you directly name it after the file system that you're parsing? Your current implementation is geared towards efivarfs. So how about you call the driver "efivarfs"?
2. Please add documentation for the new type in README.md. Add a new section about the efivarfs support and call out its limitations (see above). Make sure to warn anyone who would want to use this to transfer UEFI Secure Boot data. Ideally even add such a warning in code: If you find a protected variable, emit a warning.

File: pyuefivars/raw.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+GUID_RE = re.compile('[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}', re.I)
+
+
+class RAWUEFIVarStore(UEFIVarStore):
+
+    def __init__(self, dummy):
+        super().__init__()
+        del dummy
+        self.is_empty = False
+        if not os.path.isdir("/sys/firmware/efi") or not os.path.isdir('/sys/firmware/efi/efivars'):
```
Comment (agraf) [inline pyuefivars/raw.py:20]: I don't understand why we need this check. Is there a different sanity check we can do to check if the directory contents is sensible? I can absolutely imagine a user that tars up their efivarfs, transfers it to a different machine and then runs this tool on the extracted directory.

File: pyuefivars/raw.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            self.is_empty = True
+            return
+
+        var_names = os.listdir('/sys/firmware/efi/efivars')
```
Comment (agraf) [inline pyuefivars/raw.py:24]: Then ensure to only use the user provided path going forward

File: pyuefivars/raw.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        for var_name in var_names:
+            # print(f"working on variable {var_name}")
+            res = GUID_RE.search(var_name)
+            guid_str = var_name[res.span()[0]: res.span()[1]]
+            guid = uuid.UUID(guid_str).bytes_le
+            name = var_name[:res.span()[0] - 1]  # need to remove last "-" before GUID
+            with open(os.path.join('/sys/firmware/efi/efivars', var_name), 'rb') as f:
+                content = f.read()
+            data = content[1:]
+            attr = int.from_bytes(content[:1], "big")
```
Comment (agraf) [inline pyuefivars/raw.py:36]: Is this correct for all variables? Could you please take a UEFI Secure Boot enabled VM that is backed by either an edk2 or aws varstore, then run the new backend on that VM and generate a JSON output of that, then generate a diff between this JSON and the original converted to JSON. I'm curious to see what we're losing.

File: pyuefivars/raw.py (hunk 1 above)
Comment (umishko) [reply pyuefivars/raw.py:20]: I  have very limited time for testing so let's agree that first person that needs this functionality would add it

**Pull request closed without merging**